src/shims/unix/linux/foreign_items.rs

```diff
@@ -72,6 +72,60 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             }
 
             // Threading
+            "prctl" => {
+                // We do not use `check_shim` here because `syscall` is variadic. The argument
+                // count is checked bellow.
+                this.check_abi_and_shim_symbol_clash(abi, Abi::C { unwind: false }, link_name)?;
+
+                if args.is_empty() {
+                    throw_ub_format!(
+                        "incorrect number of arguments for prctl: got 0, expected at least 1"
+                    );
+                }
+
+                // Since only two prctl calls are currently handled, and both expext just two arguments,
+                // it makes sense to handle validation here, outside of the match block below.
+                if args.len() < 2 {
+                    throw_ub_format!(
+                        "incorrect number of arguments for `PR_SET_NAME` prctl: got {}, expected at least 2",
+                        args.len()
+                    );
+                }
+
+                for (i, arg) in args.iter().enumerate() {
+                    let val = this.read_scalar(arg)?.to_i32()?;
+                    if val != 0 {
+                        throw_ub_format!(
+                            "incorrect value of {}th argument for 'PR_SET_NAME' prctl: got {}, expected 0L",
```

Comment:
Not sure what `0L` is. We're writing Rust here, not C. :)
Why should all arguments have to be 0? This check is quite strange and seems to reject most callers...?

Reply:
I saw a lot of `prctl` calls from the C code where additional arguments were provided while they were unused. And the function documents state that it's okay.

Reply:
But you are checking that *all* arguments are 0, not just all "additional" arguments?

Anyway, please follow what `fnctl` does. It just ignores extra arguments entirely.

Reply:
Right, another thing caused by late time coding (: I will fix it when I come to the keyboard.

Reply:
Found that piece of code even in the `std` library:

```rust
#[cfg(target_os = "android")]
pub fn set_name(name: &CStr) {
    const PR_SET_NAME: libc::c_int = 15;
    unsafe {
        libc::prctl(
            PR_SET_NAME,
            name.as_ptr(),
            0 as libc::c_ulong,
            0 as libc::c_ulong,
            0 as libc::c_ulong,
        );
    }
}
```

Not required, but provided for an unknown reason.